src/client/microsoftAuth.js

```diff
@@ -11,113 +10,61 @@ const getFetchOptions = {
   }
 }
 
-/**
- * Obtains Minecaft profile data using a Minecraft access token and starts the join sequence
- * @param {object} client - The client passed to protocol
- * @param {object} options - Client Options
- * @param {string} mcAccessToken - Minecraft access token for session server
- * @param {object?} msa - Cached Microsoft account data for more descriptive errors
- */
-async function postAuthenticate (client, options, mcAccessToken, msa) {
-  options.haveCredentials = mcAccessToken != null
-
-  let minecraftProfile
-  const res = await fetch(authConstants.MinecraftServicesProfile, getFetchOptions)
-  if (res.ok) { // res.status >= 200 && res.status < 300
-    minecraftProfile = await res.json()
-  } else {
-    const user = msa ? msa.getUsers()[0] : options.username
-    throw Error(`Failed to obtain Minecraft profile data for '${user?.username}', does the account own Minecraft Java? Server returned: ${res.statusText}`)
+async function authenticate (client, options) {
+  if (!options.profilesFolder && options.profilesFolder !== false) {
+    options.profilesFolder = require('minecraft-folder-path')
   }
 
-  if (!minecraftProfile.id) {
-    debug('[mc] profile', minecraftProfile)
+  const Authflow = new PrismarineAuth(options.username, options.profilesFolder, options, options.onMsaCode)
+  const accessToken = await Authflow.getMinecraftJavaToken()
+
+  // We have a minecraft access token.
+  // Let's verify entitlements.
+
+  getFetchOptions.headers.Authorization = `Bearer ${accessToken}`
+  const MineEntitlements = await fetch(Endpoints.MinecraftServicesEntitlement, getFetchOptions).then(checkStatus)
```

Comment:
please use lower camel casing when declaring variables, upper camel case when defining classes. This makes it easier for others to differentiate.